### PR TITLE
 #160814323 enable social login

### PR DESCRIPTION
**Pull request by marthamareal**
#### What does this PR do?
Enable users to signup or login with their social accounts(google)
#### Description of Task to be completed?
- create login components and their actions
- style the login buttons
- test new functionalities
#### How should this be manually tested?
- Run the project with `npm start`
- Go to login on the menu and click the buttons
#### What are the relevant pivotal tracker stories?
- [#160814323](https://www.pivotaltracker.com/story/show/160814323)
#### Screenshots (if appropriate)
screen shot 2018-10-10 at 10 46 54

screen shot 2018-10-08 at 16 54 06

#### Questions:
(Not Applicable)

### CODE REVIEW

**Comment by cmplx-xyttmt** [comment]
good work, @marthamareal . 
LGTM

**Comment by ruganda** [comment]
Hello @marthamareal . please remove fix the failing checks on your PR.

**Comment by marthamareal** [comment]
> Hello @marthamareal . please remove fix the failing checks on your PR.

Thanks @ruganda , I have fixed them

**Comment by byarustev** [comment]
Martha thanks for the good work.  But I think putting the urls in a sigle file then just importing them would make making changes easier. 
Then since the part of the backend server url is the same I think using a variable to hold that part of the url then keep concatenating the other parts.
eg 
const SERVER_URL ="https://ah-backend-tesseract-staging.herokuapp.com/api/"
const SOCIAL_URL =SERVER_URL+"social/";

**Comment by ruganda** [comment]
LGTM

**Comment by marthamareal** [comment]
> Martha thanks for the good work. But I think putting the urls in a sigle file then just importing them would make making changes easier.
> Then since the part of the backend server url is the same I think using a variable to hold that part of the url then keep concatenating the other parts.
> eg
> const SERVER_URL ="https://ah-backend-tesseract-staging.herokuapp.com/api/"
> const SOCIAL_URL =SERVER_URL+"social/";

Thanks, @byarustev, however, need more color where would you prefer that file to be?

File: src/components/auth/SocialButtons.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  };
+  renderTwitter = () => {
+    return (
+      <TwitterLogin
+        className=" twitter waves-effect btn row light-blue"
+        id={"twitter"}
+        loginUrl="https://ah-backend-tesseract-staging.herokuapp.com/oauth/complete/twitter/"
+        onSuccess={this.onSuccess("twitter")}
+        onFailure={this.onLoginFailure}
+        requestTokenUrl="https://ah-backend-tesseract-staging.herokuapp.com/oauth/login/twitter/"
```
Comment (inno-asiimwe) [inline src/components/auth/SocialButtons.js]: Can this be made reusable? What happens when I am running this on localhost?

Comment (marthamareal) [reply src/components/auth/SocialButtons.js]: let me make it reusable

**Pull request merged**